syntax = "proto3";

package fixbug;

//定义下面的方法
option cc_generic_services = true;

message ResultCode{
    int32 errcode = 1;
    bytes errmsg = 2;
}
// 登录请求消息类型
message LoginRequest {
    bytes name = 1; // 字段 "name"，字段标签为 1
    bytes pwd = 2;  // 字段 "pwd"，字段标签为 2
    // map<int32,string> test;
}

// 登录响应消息类型
message LoginResponse {
    ResultCode result = 1 ;
    bool success = 2;    // 字段 "success"，字段标签为 3
}

message RegisterRequest{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse {
    ResultCode result = 1 ;
    bool success = 2;    // 字段 "success"，字段标签为 3
}


//在protobuf里怎么定义描述rpc方法的类型
service UserServiceRpc{
    rpc Login(LoginRequest) returns (LoginResponse);//最好与本地的方法保持一致
    rpc Register(RegisterRequest) returns (RegisterResponse);
    // rpc GetFriendList(GetFriendListRequest) returns (GetFriendListResponese);
}


